lib/pure/json.nim

```diff
@@ -1010,8 +1010,14 @@ when defined(nimFixedForwardGeneric):
       dst = T(jsonNode.num)
 
   proc initFromJson[T: enum](dst: var T; jsonNode: JsonNode; jsonPath: var string) =
-    verifyJsonKind(jsonNode, {JString}, jsonPath)
-    dst = parseEnum[T](jsonNode.getStr)
+    verifyJsonKind(jsonNode, {JInt, JString}, jsonPath)
+    if jsonNode.kind == JInt:
+      let i = jsonNode.getBiggestInt
+      if (i <= low(T).BiggestInt or i >= high(T).BiggestInt) or $i.T == $i & " (invalid data!)":
```

Comment:
The problem here is, that if you don't throw an exception, you allocate three temporary strings every time you initialize something from json from these three expressions: `$i.T`, `$i` and `$i & " (invalid data!)"`. This is a performance problem, because allocation and freeing memory is generally slow. Then relying on the stringification of invalid enums is not something that I consider as reliable as string representations might change.

Reply:
whichi is why i was asking in the outdated review above if there was a better way

Reply:
```suggestion
      const validValues = intValuesSet(T)
      if i in validValues:
```

Reply:
Please no. Don't use a set for this.